### PR TITLE
Initial DomD passthrough

### PR DESCRIPTION
**Pull request by andr2000**
Initial Dom0/DomD configuration for Salvator-X M3
    DomD configuration:
      - pass machine compatible via partial dtb
      - copy dtb nodes other than default aliases/passthrough:
        - soc
        - clocks
        - regulators
      - configure iomem/irq/dtdev
      - mark as passthrough devices in Dom0 dtb
      - remove scif2
    
Add how-to document for debugging DomD dtb
    
Add patches to Xen which allow extending partial DTB creation
    These patches allow copying additional nodes while creating
    partial device trees and specifying machine compatible
    strings.

### CODE REVIEW

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/0001-dts_dom0-Add-ranges-that-AUDMACs-relies-on.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
+From f3174055c85d01b04dbd791d90562e017684b80c Mon Sep 17 00:00:00 2001
+From: Oleksandr Tyshchenko <olekstysh@gmail.com>
+Date: Tue, 24 Jan 2017 21:06:12 +0200
+Subject: [PATCH] dts_dom0: Add ranges that AUDMACs relies on
```
Comment (otyshchenko1) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/0001-dts_dom0-Add-ranges-that-AUDMACs-relies-on.patch]: Pls clarify patch subject -> change "dts_dom0" to something like "r8a7795(6).dtsi"

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts [hunk 2]
```diff
@@ -986,14 +991,14 @@
 	status = "okay";
 };
```
Comment (otyshchenko1) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts:973]: Why IPMMUs are still enabled in domd device tree? They must be disabled or even dropped. Moreover, Xen doesn't insert IOMMU in dom0 device tree even it is not owned by Xen.

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts [hunk 3]
```diff
@@ -36,15 +36,20 @@
...
 
 	aliases {
-		serial0 = &scif2;
 		serial1 = &scif1;
 		ethernet0 = &avb;
 	};
 
+	soc {
+		//this must point to Xen interrupt-parent which is 0xfde8
+		interrupt-parent = <0xfde8>;
```
Comment (otyshchenko1) [inline recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts]: Looks like it is not a worldwide constant.  Can't we insert the proper value in an automatic way?

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd.cfg [hunk 4]
```diff
@@ -15,6 +15,258 @@ device_tree = "/xt/domd/domd.dtb"
...
+    "/x12_clk",
+    "/audio_clkout",
+    "/sound",
+    "/vga-encoder",
+    "/vga",
+    "/vspm_if",
+    "/lvds-encoder",
+    "/lvds",
+    "/hdmi0-out",
+    "/regulator-vbus0-usb2",
```
Comment (otyshchenko1) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd.cfg:65]: Please add a comment in a code explaining why we need to handle these clocks and regulators separately.

**Review by otyshchenko1 — COMMENTED** [review]
R-b with comments being taken care of.

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd.cfg (hunk 4 above)
Comment (andr2000) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd.cfg:65]: I don't know, you help me understand if we need these

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/0001-dts_dom0-Add-ranges-that-AUDMACs-relies-on.patch (hunk 1 above)
Comment (andr2000) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/0001-dts_dom0-Add-ranges-that-AUDMACs-relies-on.patch]: done

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts (hunk 3 above)
Comment (andr2000) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts]: I see no way to do so in a generic way:
tools/libxl/libxl_arm.c:
/*
 * The device tree compiler (DTC) is allocating the phandle from 1 to
 * onwards. Reserve a high value for the GIC phandle.
 */
#define PHANDLE_GIC (65000)

Comment (andr2000) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts]:
```diff --git a/recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts b/recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts
index 3316682e5616..30274ecfa76f 100644
--- a/recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts
+++ b/recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts
@@ -35,6 +35,12 @@
 #include "r8a7796.dtsi"
 #include <dt-bindings/gpio/gpio.h>
 
+/*
+ * The device tree compiler (DTC) is allocating the phandle from 1 to
+ * onwards. Reserve a high value for the GIC phandle.
+ */
+#define PHANDLE_GIC (65000)
+
 / {
 
        aliases {
@@ -43,8 +49,8 @@
        };
 
        soc {
-               //this must point to Xen interrupt-parent which is 0xfde8
-               interrupt-parent = <0xfde8>;
+               //this must point to Xen interrupt-parent
+               interrupt-parent = <PHANDLE_GIC>;
        };
 
        passthrough {
```

File: recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts (hunk 2 above)
Comment (andr2000) [reply recipes-domd/agl/files/meta-xt-prod-extra/recipes-kernel/linux/linux-renesas/r8a7796-salvator-x-domd.dts:973]: removed

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd.cfg (hunk 4 above)
Comment (andr2000) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-extended/guest-addons/files/domd.cfg:65]: removed regulators

**Pull request merged**